aws/sagemaker/large-model-inference/sample-llm/trtllm_rollingbatch_deploy_llama_13b.ipynb

```diff
@@ -0,0 +1,243 @@
+    "\n",
+    "- S3 bucket push access\n",
+    "- SageMaker access\n",
+    "\n",
+    "## Step 1: Let's bump up SageMaker and import stuff"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "67fa3208",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install sagemaker --upgrade  --quiet"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ec9ac353",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import sagemaker\n",
+    "from sagemaker import Model, image_uris, serializers, deserializers\n",
+    "\n",
+    "role = sagemaker.get_execution_role()  # execution role for the endpoint\n",
+    "sess = sagemaker.session.Session()  # sagemaker session for interacting with different AWS APIs\n",
+    "region = sess._region_name  # region name of the current SageMaker Studio environment\n",
+    "account_id = sess.account_id()  # account_id of the current SageMaker Studio environment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "81deac79",
+   "metadata": {},
+   "source": [
+    "## Step 2: Start preparing model artifacts\n",
+    "In LMI container, we expect some artifacts to help setting up the model\n",
+    "- serving.properties (required): Defines the model server settings\n",
+    "- model.py (optional): A python file to define the core inference logic\n",
+    "- requirements.txt (optional): Any additional pip wheel need to install"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b011bf5f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile serving.properties\n",
+    "engine=MPI\n",
+    "option.model_id=TheBloke/Llama-2-13B-fp16\n",
+    "option.tensor_parallel_degree=4\n",
+    "option.max_rolling_batch_size=64\n",
+    "option.rolling_batch=trtllm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b0142973",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%sh\n",
+    "mkdir mymodel\n",
+    "mv serving.properties mymodel/\n",
+    "tar czvf mymodel.tar.gz mymodel/\n",
+    "rm -rf mymodel"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e58cf33",
+   "metadata": {},
+   "source": [
+    "## Step 3: Start building SageMaker endpoint\n",
+    "In this step, we will build SageMaker endpoint from scratch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4d955679",
+   "metadata": {},
+   "source": [
+    "### Getting the container image URI\n",
+    "\n",
+    "[Large Model Inference available DLC](https://github.com/aws/deep-learning-containers/blob/master/available_images.md#large-model-inference-containers)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7a174b36",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "image_uri = \"763104351884.dkr.ecr.us-east-1.amazonaws.com/djl-inference:0.25.0-tensorrtllm0.5.0-cu122\""
```

Comment:
you need to fetch from SageMaker pySDK. Use our staging build for your test

Reply:
Done.